### PR TITLE
Document publisher properties in API docs

### PR DESCRIPTION
**Pull request by stIncMale**
JAVA-3963

### CODE REVIEW

File: driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/package-info.java [hunk 1]
```diff
@@ -15,6 +15,17 @@
  */
 
 /**
- * This packages contains classes for the reactive stream client implementation
+ * This packages contains classes for the reactive stream client implementation.
+ * <p>
+ * All API {@link org.reactivestreams.Publisher}s are
```
Comment (jyemin) [inline driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/package-info.java:20]: Do these links end up resolving correctly to some stable location for the Reactive Streams documentation?

Comment (stIncMale) [reply driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/package-info.java:20]: No, they do not. However, we have already had this situation before this PR. For example, see https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-reactivestreams/com/mongodb/reactivestreams/client/FindPublisher.html - there is no javadoc link to `org.reactivestreams.Publisher`.

I noticed the lack of links previously and saw that the only external API link we have configured is https://docs.oracle.com/en/java/javase/11/docs/api/. If there is no preventive rule, I am glad to add the link to the [`org.reactivestreams` API](https://www.reactive-streams.org/reactive-streams-1.0.2-javadoc/org/reactivestreams/package-summary.html).

I tried building javadocs with my change locally, and the change works as expected.

**Review by rozza — APPROVED** [review]
LGTM

**Review by jyemin — APPROVED** [review]
LGTM

**Pull request merged**